docker/test/integration/minifi/core/KindProxy.py

```diff
@@ -0,0 +1,112 @@
+
+
+class KindProxy:
+    REPOSITORY = 'minifi-kubernetes-test'
+    TAG = 'v1'
+
+    def __init__(self, temp_directory):
+        self.temp_directory = temp_directory
+        self.kind_binary_path = os.path.join(self.temp_directory, 'kind')
+        self.kind_config_path = os.path.join(self.temp_directory, 'kind-config.yml')
+        self.__download_kind()
+
+    def __download_kind(self):
+        if subprocess.run(['curl', '-Lo', self.kind_binary_path, 'https://kind.sigs.k8s.io/dl/v0.11.1/kind-linux-amd64']).returncode != 0:
+            raise Exception("Could not download kind")
+        os.chmod(self.kind_binary_path, stat.S_IXUSR)
+
+    def create_config(self, volumes):
+        kind_config = dedent("""\
+                apiVersion: kind.x-k8s.io/v1alpha4
+                kind: Cluster
+                nodes:
+                   - role: control-plane
+                """)
+
+        if volumes:
+            kind_config += "     extraMounts:\n"
+
+        for host_path, container_path in volumes.items():
+            kind_config += "      - hostPath: {path}\n".format(path=host_path)
+            kind_config += "        containerPath: {path}\n".format(path=container_path['bind'])
+            if container_path['mode'] != 'rw':
+                kind_config += "        readOnly: true\n"
+
+        with open(self.kind_config_path, 'wb') as config_file:
+            config_file.write(kind_config.encode('utf-8'))
+
+    def start_cluster(self):
+        subprocess.run([self.kind_binary_path, 'delete', 'cluster'])
+
+        if subprocess.run([self.kind_binary_path, 'create', 'cluster', '--config=' + self.kind_config_path]).returncode != 0:
+            raise Exception("Could not start the kind cluster")
+
+    def load_docker_image(self, image_store):
+        image_store.get_image('minifi-cpp-in-kubernetes')
+
+        if subprocess.run([self.kind_binary_path, 'load', 'docker-image', KindProxy.REPOSITORY + ':' + KindProxy.TAG]).returncode != 0:
+            raise Exception("Could not load the minifi docker image into the kind cluster")
+
+    def create_objects(self):
+        test_dir = os.environ['TEST_DIRECTORY']
+        resources_directory = os.path.join(test_dir, 'resources', 'kubernetes', 'pods-etc')
+
+        self.__wait_for_default_service_account('default')
+        namespaces = self.__create_objects_of_type(resources_directory, 'namespace')
+        for namespace in namespaces:
+            self.__wait_for_default_service_account(namespace)
+
+        self.__create_objects_of_type(resources_directory, 'pod')
+        self.__create_objects_of_type(resources_directory, 'clusterrole')
+        self.__create_objects_of_type(resources_directory, 'clusterrolebinding')
+
+    def __wait_for_default_service_account(self, namespace):
+        for _ in range(120):
+            if subprocess.run(['docker', 'exec', 'kind-control-plane', 'kubectl', '-n', namespace, 'get', 'serviceaccount', 'default']).returncode == 0:
```

Comment:
It may be better to instantiate a python docker client to get the container and call `exec_run` on it to execute commands inside the container.

Reply:
Do you know how to redirect a file to the stdin of `exec_run`?  This is needed in `__create_objects_of_type()`.

Reply:
Unfortunately I didn't find any similar functionality in `exec_run`'s documentation, I would suggest using docker copy to copy that file inside the container, but if you would like to keep it as it is due to this I'm okay with that as well.

Reply:
I have changed `subprocess.run()` to `docker_client.exec_run()` where I could in https://github.com/apache/nifi-minifi-cpp/pull/1270/commits/57741586adabccb1706313c2f353e20c769270f7.

Unfortunately, there is no API for `docker cp` in the Python client library.  I didn't like the recommended workaround of "create a tar archive, use `put_archive()` to copy it to the container, then use `exec_run()` to untar it inside", so I left that as `subprocess.run()`.

Reply:
That workaround sounds way too complicated to use, this is a better solution at the moment, thanks.